src/Tests/dotnet-publish.Tests/GivenDotnetPublishPublishesProjects.cs

```diff
@@ -137,6 +141,89 @@ public void ItPublishesSelfContainedWithRid(string args)
+            var outputDirectory = PublishApp(testAppName, rid, "-p:PublishSelfContained=true");
+
+            var outputProgram = Path.Combine(outputDirectory.FullName, $"{testAppName}{Constants.ExeSuffix}");
+
+            outputDirectory.Should().HaveFiles(new[] {
+                "System.dll", // File that should only exist if self contained 
+            });
+
+            new RunExeCommand(Log, outputProgram)
+                .Execute()
+                .Should().Pass()
+                     .And.HaveStdOutContaining("Hello World");
+        }
+
+        [Theory]
+        [InlineData(true, false, false)] // PublishSC sets SC to true even if SC is false in the project file
+        [InlineData(false, false, false)] // PublishSC sets SC to false even if SC is true in the project file 
+        [InlineData(true, true, false)] // PublishSC does not take effect if SC is global
+        public void PublishSelfContainedPropertyDoesOrDoesntOverrideSelfContained(bool publishSelfContained, bool selfContainedIsGlobal, bool publishSelfContainedIsGlobal)
+        {
+            bool selfContained = !publishSelfContained;
+            bool resultShouldBeSelfContained = publishSelfContained && !selfContainedIsGlobal;
+
+            string targetFramework = ToolsetInfo.CurrentTargetFramework;
+            var testProject = new TestProject("MainProject")
+            {
+                TargetFrameworks = targetFramework,
+                IsExe = true
+            };
+
+            testProject.RecordProperties("SelfContained");
+            if (!publishSelfContainedIsGlobal)
+                testProject.AdditionalProperties["PublishSelfContained"] = publishSelfContained.ToString();
+            if (!selfContainedIsGlobal)
+                testProject.AdditionalProperties["SelfContained"] = selfContained.ToString();
+
+            var testAsset = _testAssetsManager.CreateTestProject(testProject, identifier: $"PSC-OVERRIDES-{publishSelfContained}-{selfContainedIsGlobal}-{publishSelfContainedIsGlobal}");
+            var publishCommand = new DotnetCommand(Log);
+            List<string> args = new List<string>
+            {
+                "publish",
+                selfContainedIsGlobal ? $"/p:SelfContained={selfContained}" : "",
+                publishSelfContainedIsGlobal ? $"/p:PublishSelfContained={publishSelfContained}" : "",
+            };
+
+            publishCommand
+                .WithWorkingDirectory(Path.Combine(testAsset.Path, "MainProject"))
+                .Execute(args.ToArray())
+                .Should()
+                .Pass();
+
+            var publishedDirectory = testAsset.TestRoot;
+            publishedDirectory = Path.Combine(publishedDirectory, "MainProject", "obj", "Debug", targetFramework);
+            if (resultShouldBeSelfContained)
+                publishedDirectory = Directory.GetDirectories(publishedDirectory).FirstOrDefault(); // get the rid in the directory
```

Comment:
If there is more than one RID directory, then it would be better to throw an exception here rather than to return null and fail later.

```suggestion
                publishedDirectory = Directory.GetDirectories(publishedDirectory).First(); // get the rid in the directory
```

Reply:
Good point, thanks. With the below change this won't be needed.